xonsh/commands_cache.py

```diff
@@ -168,8 +168,10 @@ def _update_paths_cache(self, paths: tp.Sequence[str]) -> bool:
         updated = False
         for path in paths:
             modified_time = os.stat(path).st_mtime
-            if (path not in self._paths_cache) or (
-                self._paths_cache[path].mtime != modified_time
+            if (
+                self.env.get("ENABLE_COMMANDS_CACHE", True)
```

Comment:
the condition is the exact opposite. currently if its enabled, it means to disable